### PR TITLE
feat(RE-1033):  Added projects configmap and job

### PR DESCRIPTION
**Pull request by vinayp-ops**
#RE-1033
This PR contains the configMap and Job to create project on upgrade of the chart.

### CODE REVIEW

File: templates/custom/projects-cm.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      -u $basicAuthCreds)
+      echo "response: " $res
+    }
+
+    function createProject() {
+      # call Project API and create the project with the values taken from the values yaml
+      project=$(curl -X POST "$coreHostUrl/api/projects" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_ADMIN_USERNAME}":"${HARBOR_PASSWORD}" \
```
Comment (agill17) [inline templates/custom/projects-cm.yaml]: replace with `basicAuthCreds`

File: templates/custom/projects-cm.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    function updateProjectWithMemberGroup() {
+      projectName=$1
+      name=$(IFS=$',' && echo $1 | awk -F '-' '{print $1}')
+      role_id=$(IFS=$',' && echo $1 | awk -F '-' '{print $2}')
+
+      # Get projecid for the given projectname.
+      projectID=$(curl -X GET "$coreHostUrl/api/projects?name=$projectName" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_USER_NAME}":"${HARBOR_ADMIN_PASSWORD}"  | jq '.[].project_id ')
```
Comment (agill17) [inline templates/custom/projects-cm.yaml]: replace with `basicAuthCreds`

File: templates/custom/projects-cm.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      -u "${HARBOR_USER_NAME}":"${HARBOR_PASSWORD}" \
+      | jq '.[] | map(select(.group_name == $name) | .id)'
+      echo "Member ID : ${memberID}")
+
+      # Updating project with member id
+      echo "Updating project ${projectID} with ${memberID} "
+      id=$(curl -X PUT "$coreHostUrl/api/projects/${projectID}/members/${memberID}" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_USER_NAME}":"${HARBOR_PASSWORD}" \
```
Comment (agill17) [inline templates/custom/projects-cm.yaml]: replace with basicAuthCreds

File: templates/custom/projects-cm.yaml (hunk 1 above)
Comment (vinayp-ops) [reply templates/custom/projects-cm.yaml]: Resolved and use variable 'basicAuthCreds'

File: templates/custom/projects-cm.yaml (hunk 2 above)
Comment (vinayp-ops) [reply templates/custom/projects-cm.yaml]: Resolved and use variable 'basicAuthCreds'

File: templates/custom/projects-cm.yaml (hunk 3 above)
Comment (vinayp-ops) [reply templates/custom/projects-cm.yaml]: Resolved and use variable 'basicAuthCreds'

**Comment by agill17** [comment]
Could you add sample values in values file? But have them commented out, and add comments for each value defining what they do / what they mean

File: values.yaml [hunk 4]
```diff
@@ -690,4 +690,5 @@ custom:
         ## Optional. Job annotations
         # annotations: {}
         ## Optional. Job pod annotations
-        # podAnnotations: {}
\ No newline at end of file
+        # podAnnotations: {}
+  #projects: []
```
Comment (agill17) [inline values.yaml]: add docs on what fields are available in projects and their valid values. Refer to the above custom paths for doc reference

**Review by agill17 — CHANGES_REQUESTED** [review]

File: values.yaml (hunk 4 above)
Comment (vinayp-ops) [reply values.yaml]: Updated values.yaml

**Review by agill17 — APPROVED** [review]

**Pull request merged**